src/Services/Documents.API/Documents.Application/Services/Implementations/DocumentService.cs

```diff
@@ -34,16 +34,11 @@ public async Task<Document> CreateAsync(Guid resultId, IFormFile documentFile)
 			throw new BadRequestException("Invalid document format");
 		}
 
-		var containerClient = _blobServiceClient.GetBlobContainerClient(DocumentsContainerName);
-		await containerClient.CreateIfNotExistsAsync(publicAccessType: Azure.Storage.Blobs.Models.PublicAccessType.Blob);
-
 		var filename = _filenameGenerator.Generate(documentFile.FileName);
-		var blobClient = containerClient.GetBlobClient(filename);
+		var fileUri = await _documentsContainer.UploadAsync(documentFile, filename);
 
-		var url = new Uri(blobClient.Uri, filename);
+		var url = new Uri(fileUri, filename);
 		var document = new Document(url.ToString(), resultId);
-
-		await UploadFileAsync(blobClient, documentFile);
 		await _unitOfWork.DocumentRepository.CreateAsync(document);
```

Comment:
Lets say, what if something will happen when we are trying to update EF context, so we will have to deal with situation like:
* Blob was created, but document wasnt

Looks like we have to implement some kind of transaction here =)

Reply:
Update\Delete methods as well